src/test_c2s_srv.c

```diff
@@ -280,9 +285,8 @@ int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
         log_println(0, "C2S test error: can't create pipe.");
       } else {
         if ((c2s_childpid = fork()) == 0) {
-          /* close(ctlsockfd); */
           close(testOptions->c2ssockfd);
-          close(recvsfd);
+          close(c2s_conn.socket);
```

Comment:
The s2c and c2s close() operations are close() after a fork().  They are closing a resource that the child process should not use, rather than trying to shutdown the connection.

The sfw operations I changed.